feat: add post summary micro variant

dancormier opened the pull request:
resolves #966

See https://deploy-preview-967--stacks.netlify.app/product/components/post-summary/#micro

## Questions

- **Should we truncate titles? If so, how many lines?** (This _could_ be handled in core as needed. Curious if we should prescribe truncation though. There's implications when it comes to vertically aligning elements within the post summary.)
- **What's the recommended title font size?** (Profile currently uses `body2`, sidebar currently uses `body1`)
- **Should items include borders?** (I'm assuming they can be added as needed by consumers but I'm not sure if we want to unify this.)
- **Should we include support for timestamps? Or any other optional data, for that matter?** (Profile currently includes timestamp)

## TODOs

- [ ] Update documentation

## Screenshot

### Second draft

Below are screenshots with a couple variations (unanswered/short title/no date/no menu, longer title/short date/longest title/long date). There's more that _could_ exist in these post summaries.

![image](https://user-images.githubusercontent.com/647177/171288941-4a15fab6-6642-42ee-a1cd-d2f1f6adb06e.png)

![image](https://user-images.githubusercontent.com/647177/171288883-2c8d2b85-45a1-4d45-bd1c-c30279b3cfad.png)

<details>
<summary>First draft</summary>

![image](https://user-images.githubusercontent.com/647177/171277191-f33ca971-954f-484c-97d7-d88d9ebe2558.png)

</details>

dancormier committed:
feat: add post summary micro variant

resolves #966

paintedbicycle commented:
- I think body1 seems fine
- Yes, the component should handle truncation
- Do we need the `Q` and `A` icon back? And the Article and Collection icons?

dancormier committed:
Micro variant refinements

Partially addresses https://github.com/StackExchange/Stacks/pull/967#issuecomment-1142609620

dancormier committed:
Exclude menu button from simplest example

dancormier committed:
Refine padding, menu button placement

paintedbicycle commented:
Looking at the profile page I think we need the icons, otherwise everything would look exactly the same. This could then also be used in places like the sidebar and in search like below

<img width="837" alt="Screen Shot 2022-06-02 at 8 33 43 AM" src="https://user-images.githubusercontent.com/371114/171666138-be1a0f86-a2a3-4940-8896-b0629e7987e7.png">

dancormier commented:
I'm closing this PR since it will have many, _many_ conflicts once https://github.com/StackExchange/Stacks/pull/1091 is merged. We shall revisit post-PPCP refactor.
